### PR TITLE
🎨  Move settings cache & cleanup settings API

### LINKED ISSUE #8037: Utils.url has a circular dependency
Running madge on ghost core (master) shows a circular dependency:

* `npm install -g madge`
* `madge --circular ./core/server` 

results in:

api/settings -> api/utils -> permissions/index -> models/index -> models/base/index -> utils/index -> utils/url

The problem is that api.settings ends up requiring utils/url through a complex chain. However utils/url uses api.settings to get hold of the permalink and timezone settings it needs in order to calculate urls.

FUN 😁

There is currently a TODO in the codebase to move the settingsCache.

I'm working on removing settings.availableThemes.

This is a note to remember to check this circular dep is gone before we launch.

### PR DESCRIPTION
**Pull request by ErisDS**
closes #8037

This resolves the issue with circular dependency that was killing me when trying to separate the settings / config & available themes.

I have done 3 things:

1. removed the API level default settings population. This complex code cannot have been called anywhere, as we always do the population on boot.

2. created an initialisation step, so that we can reduce the number of references to `updateSettingsCache`

3. moved the settings cache out of the API into it's own thing

With this PR in place, I have a 2nd one coming to remove the availableThemes concept from both settings and config.

### CODE REVIEW

File: core/server/settings/cache.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                debug('Auto updating', settingModel.get('key'));
+                self.set(settingModel.get('key'), settingModel.toJSON());
+            };
+        // First, reset the cache
+        settingsCache = {};
+
+        // Bind to events to automatically keep up-to-date
+        events.on('settings.edited', updateSettingFromModel);
+        events.on('settings.added', updateSettingFromModel);
+        events.on('settings.deleted', updateSettingFromModel);
```
Comment (kirrg001) [inline core/server/settings/cache.js]: This is cool 👍 

I think we don't need this code anymore:
https://github.com/ErisDS/Ghost/blob/settings-overhaul/core/server/api/themes.js#L98
^ `themeUtils.load` calls `updateSettingsCache` already - so it's called twice?

https://github.com/ErisDS/Ghost/blob/settings-overhaul/core/server/api/db.js#L67
^ `Settings.edit` triggers an event to update the cache

Maybe you know already.

Comment (ErisDS) [inline core/server/settings/cache.js]: On line 98 of the themes api the RESULT from updateSettingsCache is depended upon. So yes, I need to remove this code, but happens in the next PR after I remove that requirement as well.

I haven't done any tests around the importer - but yes, that needs to be removed as well. I will add this as a TODO for the next PR, if that is OK?

Comment (kirrg001) [inline core/server/settings/cache.js]: yeah sure 👍

**Review by kirrg001 — APPROVED** [review]
This looks great 👍

**Pull request merged**